### PR TITLE
refactor: fetch file tree

### PR DESCRIPTION
**Pull request by hacdias**
This is a little improvement to reduce the amount of copied code. This way `doFetchFileTree` opportunistically determines whether to user's IPFS Daemon or the Public Gateway. It can still be improved a little.

License: MIT
Signed-off-by: Henrique Dias <hacdias@gmail.com>

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -17,22 +17,28 @@ export class App extends Component {
...
   }
 
   componentWillMount () {
     this.props.doInitIpfs()
   }
 
   render () {
-    const Page = this.props.route
+    const { route: Page, ipfsReady, ipfsInitFailed } = this.props
+    // Only shows the page if IPFS is ready of if the initialization has failed.
```
Comment (fsdiogo) [inline src/App.js]: typo: `of` -> `or`

File: src/App.js [hunk 2]
```diff
@@ -17,22 +17,28 @@ export class App extends Component {
...
+    // This way we can make sure we always use user's own node if available and
+    // the public gatway otherwise.
```
Comment (fsdiogo) [inline src/App.js]: typo: `gatway` -> `gateway

File: src/bundles/files.js [hunk 3]
```diff
@@ -222,61 +219,42 @@ export default {
   },
 
   doFetchFileTree: (hash) => async ({ dispatch, store, getIpfs }) => {
-    const ipfs = getIpfs()
+    let ifpsFiles
```
Comment (fsdiogo) [inline src/bundles/files.js]: `ifpsFiles` -> `ipfsFiles` 😄

Comment (fsdiogo) [reply src/bundles/files.js]: Replace all, there are more of those!

File: src/App.js [hunk 4]
```diff
@@ -17,22 +17,28 @@ export class App extends Component {
...
+    const ready = ipfsReady || ipfsInitFailed
 
     return (
       <div className='App sans-serif' onClick={navHelper(this.props.doUpdateUrl)}>
         <div className='flex flex-column min-vh-100'>
           <Header />
           <main className='flex-auto pa4'>
-            <Page />
+            { ready && <Page /> }
```
Comment (fsdiogo) [inline src/App.js]: Not sure about this, if it takes a bit to init ipfs the page will be blank with nothing to show. Do you think this is the best solution?

**Comment by fsdiogo** [comment]
I think it would be helpful to know in the dispatched action if it's fetching the files tree from the gateway no?

File: src/App.js (hunk 4 above)
Comment (hacdias) [reply src/App.js]: @fsdiogo we're just using window.ipfs or js-ipfs-api and then requesting the ID. It should not take much time, but we can add a loading animation. What do you think?

**Comment by hacdias** [comment]
@fsdiogo we are dispatching all the actions as `FILES_FETCH_*`. Previously it had `FILES_FETCH_GATEWAY_*` too but it didn't add any more information to state than `FILES_FETCH_*`'s actions.

File: src/App.js (hunk 4 above)
Comment (fsdiogo) [reply src/App.js]: Alright, we can do that later in another PR!

**Comment by fsdiogo** [comment]
@hacdias next time please squash the PR commits into one, it keeps the git log cleaner in the master branch and groups the work better 🙂

**Comment by hacdias** [comment]
@fsdiogo that was what I initially wanted to do but I couldn't change the button to Squash and Merge for some reason. Maybe something on my end

**Comment by fsdiogo** [comment]
No sweat 👍

**Pull request merged**